File: dont-run.py
from multiprocessing import Process, cpu_count
import time

def count(num):
    while num > 0:
        num -= 1
        time.sleep(0.5)
        
def spawn(num_pro):
    processes = [Process(target = count, args=(1000,)) for i in range(num_pro)]
    for p in processes:
        p.start()
        print(f"Started Process: {p.pid}")
    for p in processes:
        p.join()
        print(f"Process {p.pid} has finished!")
        
def main():
    processors = cpu_count()
    print("Number of Processors: ",processors)
    processes = processors * 300
    print(f"Creating {processes} Processes")

    while True:
        spawn(processes)    

    spawn(processes)

if __name__ == "__main__":
    main()

File: dont-run.js
const { Worker } = require('worker_threads');
const os = require('os');

function count() {
    let num = 1000;
    while (num > 0) {
        num -= 1;
        // Simulate work
        for (let i = 0; i < 1000000; i++) {}
    }
}

function spawn(numWorkers) {
    const workers = [];
    
    for (let i = 0; i < numWorkers; i++) {
        const worker = new Worker(`
            const { parentPort } = require('worker_threads');
            let num = 1000;
            while (num > 0) {
                num -= 1;
                // Simulate work
                for (let i = 0; i < 1000000; i++) {}
            }
            parentPort.postMessage('done');
        `, { eval: true });
        
        workers.push(worker);
        console.log(`Started Worker: ${worker.threadId}`);
        
        worker.on('message', (msg) => {
            console.log(`Worker ${worker.threadId} has finished!`);
        });
    }
}

function main() {
    const cpus = os.cpus().length;
    console.log("Number of CPUs:", cpus);
    const workers = cpus * 300;
    console.log(`Creating ${workers} Workers`);

    while (true) {
        spawn(workers);
    }
}

main();
